File: app/src/main/java/com/example/proyectoandroidtdp/Filtros/CreadorDeFiltros.java
package com.example.proyectoandroidtdp.Filtros;

import android.text.InputFilter;
import android.text.Spanned;

public class CreadorDeFiltros implements CreadorDeFiltrosAbstracto{

    public CreadorDeFiltros(){
    }


    public InputFilter getFiltroBinarioFraccionario(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i)<'.' || source.charAt(i)>'1') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroBinarioFraccionarioSignado(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i)<'-' || source.charAt(i)>'1') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroOctalFraccionario() {
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i) < '.' || source.charAt(i) > '7') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroDecimalFraccionario(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i)<'.' || source.charAt(i)>'9') {
                        return "";
                    }
                }
                return null;
            }
        };
    }



    public InputFilter getFiltroHexaFraccionario(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
                source = source.toString().toUpperCase();

                for (int i = start; i < end; i++) {
                    if (!(source.charAt(i)=='.' || Character.isDigit(source.charAt(i)) ||
                            (source.charAt(i)>='A' && source.charAt(i)<='F') ||
                            (source.charAt(i)>='a' && source.charAt(i)<='f'))) {
                        return "";
                    }
                }
                return source;
            }
        };
    }


    public InputFilter getFiltroBinarioEntero(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i)<'0' || source.charAt(i)>'1') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroOctalEntero() {
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i) < '0' || source.charAt(i) > '7') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroDecimalEntero(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {

                for (int i = start; i < end; i++) {
                    if (source.charAt(i)<'0' || source.charAt(i)>'9') {
                        return "";
                    }
                }
                return null;
            }
        };
    }


    public InputFilter getFiltroHexaEntero(){
        return new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
                source = source.toString().toUpperCase();

                for (int i = start; i < end; i++) {
                    if (!(Character.isDigit(source.charAt(i)) ||
                            (source.charAt(i)>='A' && source.charAt(i)<='F') ||
                            (source.charAt(i)>='a' && source.charAt(i)<='f'))) {
                        return "";
                    }
                }
                return source;
            }
        };
    }


}

File: app/src/main/java/com/example/proyectoandroidtdp/filtros/CreadorDeFiltros.kt
package com.example.proyectoandroidtdp.filtros

import android.text.InputFilter

class CreadorDeFiltros : CreadorDeFiltrosAbstracto {

    override val filtroBinarioFraccionario: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '.' || source[i] > '1') {
                    return@InputFilter ""
                }
            }
            null
        }
    override val filtroBinarioFraccionarioSignado: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '-' || source[i] > '1') {
                    return@InputFilter ""
                }
            }
            null
        }

    override val filtroOctalFraccionario: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '.' || source[i] > '7') {
                    return@InputFilter ""
                }
            }
            null
        }


    override val filtroDecimalFraccionario: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '.' || source[i] > '9') {
                    return@InputFilter ""
                }
            }
            null
        }


    override val filtroHexaFraccionario: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            val sourceUpper = source.toString().toUpperCase()
            for (i in start until end) {
                if (!(sourceUpper[i] == '.' || Character.isDigit(sourceUpper[i]) ||
                                sourceUpper[i] in 'A'..'F' ||
                                sourceUpper[i] in 'a'..'f')) {
                    return@InputFilter ""
                }
            }
            sourceUpper
        }
    override val filtroBinarioEntero: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '0' || source[i] > '1') {
                    return@InputFilter ""
                }
            }
            null
        }
    override val filtroOctalEntero: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '0' || source[i] > '7') {
                    return@InputFilter ""
                }
            }
            null
        }
    override val filtroDecimalEntero: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            for (i in start until end) {
                if (source[i] < '0' || source[i] > '9') {
                    return@InputFilter ""
                }
            }
            null
        }
    override val filtroHexaEntero: InputFilter?
        get() = InputFilter { source, start, end, _, _, _ ->
            var sourceUpper = source.toString().toUpperCase()
            for (i in start until end) {
                if (!(Character.isDigit(sourceUpper[i]) ||
                                sourceUpper[i] in 'A'..'F' ||
                                sourceUpper[i] in 'a'..'f')) {
                    return@InputFilter ""
                }
            }
            source
        }
}